src/main/java/mate/academy/spring/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,49 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.model.MovieSession;
+import mate.academy.spring.model.ShoppingCart;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.response.ShoppingCartResponseDto;
+import mate.academy.spring.service.MovieSessionService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+    private final DtoResponseMapper<ShoppingCartResponseDto, ShoppingCart> cartDtoResponseMapper;
+    private final UserService userService;
+    private final MovieSessionService movieSessionService;
+
+    public ShoppingCartController(ShoppingCartService shoppingCartService,
+                                  DtoResponseMapper<ShoppingCartResponseDto, ShoppingCart>
+                                          cartDtoResponseMapper,
+                                  UserService userService,
+                                  MovieSessionService movieSessionService) {
+        this.shoppingCartService = shoppingCartService;
+        this.cartDtoResponseMapper = cartDtoResponseMapper;
+        this.userService = userService;
+        this.movieSessionService = movieSessionService;
+    }
+
+    @GetMapping("by-user")
+    public ShoppingCartResponseDto getCartByUser(@RequestParam Long userId) {
```

Comment:
```suggestion
    public ShoppingCartResponseDto getByUser(@RequestParam Long userId) {
```
keep to consistency with service layer method